lib/stores/filesystem.js

```diff
@@ -253,9 +253,18 @@ class FilesystemStore {
         if (err) return done(err.code === "ENOENT" ? null : err);
         const object = new S3Object(bucket, key, content, metadata);
         if (options && (options.start || options.end)) {
+          const bytesRead = content.bytesRead;
+          const totalBytes = metadata["content-length"];
+          if (
+            bytesRead === 0 &&
+            (options && options.start && options.start >= totalBytes)
+          ) {
+            return done("ERANGE", totalBytes);
+          }
+
           object.range = {
             start: options.start || 0,
-            end: options.end || object.size - 1
```

Comment:
I think this broke the range response header in 2.2.8, but looks like it's fixed on master.